var/spack/repos/builtin/packages/templight/package.py

```diff
@@ -0,0 +1,166 @@
+    depends_on('python')
+    depends_on('py-lit', type=('build', 'run'))
+
+    def patch(self):
+        # We start with the templight source tree and an "llvm" subdir.
+        # But we actually need an llvm source tree with a "templight" subdir.
+        # Let's flip the directory organization around
+        templight_files = os.listdir('.')
+        templight_files.remove('llvm')
+        templight_dir = 'llvm/tools/clang/tools/templight'
+        os.mkdir(templight_dir)
+        for name in templight_files:
+            os.rename(name, os.path.join(templight_dir, name))
+        for name in os.listdir('llvm'):
+            os.rename(os.path.join('llvm', name), name)
+        os.rmdir('llvm')
+
+        # Tell the clang build system that it needs to build templight
+        with open("tools/clang/tools/CMakeLists.txt", "a") as cmake_lists:
+            cmake_lists.write("add_clang_subdirectory(templight)")
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.append_flags('CXXFLAGS', self.compiler.cxx11_flag)
+        run_env.set('CC', join_path(self.spec.prefix.bin, 'templight'))
+        run_env.set('CXX', join_path(self.spec.prefix.bin, 'templight++'))
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Templight is a debugging tool, not a production compiler, so we only
+        # need a very bare-bones build of clang
+        #
+        # Minimal build config ideas were taken from the llvm package, with
+        # the templight-specific assumption that we will always be building
+        # for LLVM / Clang 5.0+ and can safely ignore older tricks.
+        #
+        cmake_args = [
+            '-DLLVM_REQUIRES_RTTI:BOOL=ON',
+            '-DCLANG_DEFAULT_OPENMP_RUNTIME:STRING=libomp',
+            '-DPYTHON_EXECUTABLE:PATH={0}'.format(spec['python'].command.path),
+            '-DLLVM_EXTERNAL_POLLY_BUILD:Bool=OFF',
+            '-DLLVM_TOOL_POLLY_BUILD:Bool=OFF',
+            '-DLLVM_POLLY_BUILD:Bool=OFF',
+            '-DLLVM_POLLY_LINK_INTO_TOOLS:Bool=OFF',
+            '-DLLVM_EXTERNAL_LLDB_BUILD:Bool=OFF',
+            '-DLLVM_TOOL_LLDB_BUILD:Bool=OFF',
+            '-DLLVM_TOOL_LLD_BUILD:Bool=OFF',
+            '-DLLVM_EXTERNAL_LIBUNWIND_BUILD:Bool=OFF',
+            '-DLLVM_EXTERNAL_LIBCXX_BUILD:Bool=OFF',
+            '-DLLVM_EXTERNAL_LIBCXXABI_BUILD:Bool=OFF',
+            '-DLLVM_EXTERNAL_COMPILER_RT_BUILD:Bool=OFF',
+        ]
+
+        targets = ['NVPTX', 'AMDGPU']
+
+        if 'x86' in spec.architecture.target.lower():
+            targets.append('X86')
+        elif 'arm' in spec.architecture.target.lower():
+            targets.append('ARM')
+        elif 'aarch64' in spec.architecture.target.lower():
+            targets.append('AArch64')
+        elif 'sparc' in spec.architecture.target.lower():
+            targets.append('Sparc')
+        elif ('ppc' in spec.architecture.target.lower() or
+              'power' in spec.architecture.target.lower()):
```

Comment:
This conditional is taken from the llvm package. Flake8 is not happy about the first line ending with "or", I suppose that lint was introduced after the LLVM package was written. What is the new recommended style for this?

Reply:
Yeah, you might need to update to a newer version of flake8. They keep changing it on us.